src/app/infokit/infokit.pipe.ts

```diff
@@ -0,0 +1,51 @@
+import { Pipe, PipeTransform} from '@angular/core';
+
+@Pipe ({
+  name: 'infokitPipe'
+})
+
+export class InfokitPipe implements PipeTransform {
+  transform(dict: any[], args: any= []) {
+    let activity: any[] = [];
+    for (let key in dict) {
```

Comment:
use `.forEach` (whatever ts equivalent is) so you don't have to check for `hasOwnProperty`